CodeSnippetsReflection.OpenAPI/LanguageGenerators/PowerShellGenerator.cs

```diff
@@ -0,0 +1,307 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using CodeSnippetsReflection.StringExtensions;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Services;
+
+namespace CodeSnippetsReflection.OpenAPI.LanguageGenerators
+{
+    public class PowerShellGenerator : ILanguageGenerator<SnippetModel, OpenApiUrlTreeNode>
+    {
+        private const string requestBodyVarName = "params";
+        private const string modulePrefix = "Microsoft.Graph";
+        private const string authModuleName = modulePrefix + ".Authentication";
+        private IList<PowerShellCommandInfo> psCommands = default;
+        private Regex meSegmentRegex = new Regex("^/me($|(?=/))", RegexOptions.Compiled);
+        public PowerShellGenerator()
+        {
+            if (psCommands == default)
+            {
+                string authModulePath = default;
+                // Load MgCommandMetadata file.
+                var psModuleInstallPaths = new[] {
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "PowerShell\\Modules"),
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "powershell\\7\\Modules")
+                };
+                foreach (string modulePath in psModuleInstallPaths) {
+                    if (Directory.Exists($"{modulePath}\\{authModuleName}"))
+                        authModulePath = Directory.GetDirectories($"{modulePath}\\{authModuleName}").Max();
+                }
+                if (authModulePath == default)
+                    throw new ArgumentNullException("Microsoft.Graph PowerShell SDK could not be found on this machine. Please install the SDK using 'Install-Module Microsoft.Graph'.");
+                string MgCommandMetadataPath = Directory.GetFiles($"{authModulePath}/custom/common", "MgCommandMetadata.json").FirstOrDefault();
+                string jsonString = File.ReadAllText(MgCommandMetadataPath);
+                psCommands = JsonSerializer.Deserialize<IList<PowerShellCommandInfo>>(jsonString);
+            }
+        }
+        public string GenerateCodeSnippet(SnippetModel snippetModel)
+        {
+            var indentManager = new IndentManager();
+            var snippetBuilder = new StringBuilder();
+
+            string additionalKeySegmentParmeter = default;
+            var path = snippetModel.EndPathNode.Path.Replace("\\", "/");
+            if (meSegmentRegex.IsMatch(path))
+            {
+                path = meSegmentRegex.Replace(path, "/users/{user-id}");
```

Comment:
```suggestion
                path = meSegmentRegex.Replace(path, "/users/{");
```

In case the id parameter name is different for some reason.

Reply:
This is just us replacing `/me` segment with a `/users/[user-id}`. The matching logic will in `GetCommandForRequest` method will handle all this.